### PR TITLE
Wait for traefik before calling get_host()

### PR DESCRIPTION
**Pull request by jandubois**
On Windows get_host() gets the IP address from the traefik service, and will fail with an empty return value if the service is not yet ready.

The assert failures in get_host() don't cause a direct test failure if $(get_host) is just interpolated into a string, so it needs to be called once by itself.

Sample failure that should be fixed by this: https://github.com/rancher-sandbox/rancher-desktop/actions/runs/9104774312/job/25029228803

Note that the `wasm.bats` test already had this check: https://github.com/rancher-sandbox/rancher-desktop/blob/7bffdd78a78f664ac9565025a552e68d43a4f2de/bats/tests/k8s/wasm.bats#L105-L107

### CODE REVIEW

File: bats/tests/k8s/spinkube-npm.bats
Comment (Nino-K) [inline bats/tests/k8s/spinkube-npm.bats:59]: I know the todo was not added as part of this PR however, it is vague and is it still needed? or perhaps, maybe a link to an issue would be helpful.

**Review by Nino-K — DISMISSED** [review]

File: bats/tests/k8s/spinkube-npm.bats
Comment (jandubois) [reply bats/tests/k8s/spinkube-npm.bats:47]: I just think adding an ingress controller is overkill (and a lot of source lines to review) when a simple port-forward is enough to check the pod. It simplifies the test.

The only reason I used Traefik is because the port forwarding REST API has various issues. I thought I filed a new GitHub issue about it, but can't find it right now. Will file one later.

File: bats/tests/helpers/kubernetes.bash
Comment (jandubois) [reply bats/tests/helpers/kubernetes.bash:69]: We still use traefik, we just don't use the IP address of the service because it is the internal Slirp address and not routable from the host (because we are running non-admin installs).

So we don't need traefik to be ready to call `traefik_hostname`, but we will still need it to be ready before the `curl` can succeed.

So to keep the tests simpler I would unconditionally wait for traefik to be ready. If you are concerned about saving some time and running stuff in parallel, then we could move the `wait_for_traefik` step just before we call `traefik_hostname` the first time, but I don't think it makes much difference.

I also expect to remove much of the traefik use and replace it with direct port forwarding in the future.

Comment (jandubois) [reply bats/tests/helpers/kubernetes.bash:69]: Either way, the `wait_for_traefik` function should wait unconditionally, so it always does what the function name implies. Any conditions need to be applied on the call site.

**Review by mook-as — APPROVED** [review]
~Presumably we're also waiting for Nino's approval (since you requested review from two people…)~

Nevermind, it was just me…

**Pull request merged**
